compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -148,6 +148,13 @@ where
 /// converted rather than on `DiagArgValue`, which enables types from other `rustc_*` crates to
 /// implement this.
 pub trait IntoDiagArg {
+    /// Convert `Self` into a `DiagArgValue` suitable for rendering in a diagnostic.
+    ///
+    /// It takes a `path` where "long values" could have been written to, if the `DiagArgValue`
```

Comment:
"could have been written to" implies that if things are written to that path, that happened *before* this method is called. Is that correct? I assume that the writing happens as part of this method, so then it should say something more like "can be written to".